File: java/src/main/java/com/example/hexagonal/application/port/dto/DtoMapper.java
package com.example.hexagonal.application.port.dto;

import com.example.hexagonal.application.domain.Product;
import org.springframework.stereotype.Component;

@Component
public class DtoMapper {

  public ProductDto toDto(Product entity) {
    return ProductDto.builder()
        .id(entity.getId())
        .name(entity.getName())
        .stock(entity.getStock())
        .build();
  }
}

File: kotlin/src/main/kotlin/com/example/hexagonal/application/port/dto/DtoMapper.kt
package com.example.hexagonal.application.port.dto

import com.example.hexagonal.application.domain.Product
import org.springframework.stereotype.Component

@Component
class DtoMapper {
    fun toDto(entity: Product): ProductDto =
        ProductDto(
            id = entity.id,
            name = entity.name,
            stock = entity.stock,
        )
}
